harvard-lts/drs-import-management-service
# Advanced ingest POST controller

**Advanced ingest POST controller**
* * *

**GitHub Issue**: [(link)](https://github.com/harvard-lts/HDC/issues/105)

* Other Relevant Links [API & Queue Message Definitions](https://docs.google.com/document/d/1Q3KqCRlmcyrQfQxfCYWLUAKSs9DTOj9yp3m4ovrRA7w/edit#heading=h.hreo56n7vc4a)

# What does this Pull Request do?
Evolves the ingest POST controller to provide the necessary ingest data through the request body to initiate an ingest.

Both the request body and request response are based on the API & Queue Message Definitions document.

At this point, the object_urn field is not included in the ingest response body, since at the moment the way to obtain the value of this field is unknown.

# How should this be tested?

A description of what steps someone could take to:
* Create/Update .env and .test.env files as described in README.md
* Run the local development environment: `docker-compose -f docker-compose-dev.yml up`
* POST to https://localhost:10580/ingest with the following test request body:
`{
     "package_id": "test",
      "s3_path": "test",
     "s3_bucket_name": "test",
     "admin_metadata":
          {
             "accessFlag": "N",
              "contentModel": "opaque",
              "depositingSystem": "Harvard Dataverse",
              "firstGenerationInDrs": "unspecified",
              "objectRole": "CG:DATASET",
              "usageClass": "LOWUSE",
              "storageClass": "AR",
              "ownerCode": "123",
              "billingCode": "456",
              "resourceNamePattern": "pattern",
              "urnAuthorityPath": "path",
              "depositAgent": "789",
              "depositAgentEmail": "someone@mailinator.com",
              "successEmail": "winner@mailinator.com",
              "failureEmail": "loser@mailinator.com",
              "successMethod": "method",
              "adminCategory": "root"
          }
}`
* Check that the response to the previous request is the following:
`{
    "message": "Added to Queue",
    "package_id": "test_package_id",
    "status": "pending",
    "status_code": null
}`
* Run the local test environment: `docker-compose -f docker-compose-test.yml up`
* Execute tests: `docker exec -it test_runner pytest test`

# Test coverage
- unit tests: Yes
- integration tests: No

# Interested parties
@jessjass @ives1227
